config/application.rb

```diff
@@ -28,20 +28,30 @@ class Application < Rails::Application
     config.cli_log = 'log/cli.log'
     config.permissions_passed = false
 
-    # External configs: config.yml
-    ## Engines to load
-    ## The order determines the menu order - engines with no UI should be last
-    config.engines = config_for(:config)[:engines]
     ## Application-level menu entries - these come before Engine UIs
     config.menu_entries = [
       { caption: 'Login', icon: 'login', target: '/' },
       { caption: 'Welcome', icon: 'home', target: '/welcome' }
     ]
-    # Rancher source - for _helm_
-    config.x.rancher = OpenStruct.new(config_for(:config)[:rancher])
-    # kubernetes version
-    config.x.rancher_on_eks = OpenStruct.new(config_for(:config)[:rancher_on_eks])
-  end
+
+    # External configs: config.yml
+    begin
+      ## Engines to load
+      ## The order determines the menu order - engines with no UI should be last
+      config.engines = config_for(:config)[:engines]
+
+      # Rancher source - for _helm_
+      config.x.rancher = OpenStruct.new(config_for(:config)[:rancher])
+      # kubernetes version
+      config.x.rancher_on_eks = OpenStruct.new(config_for(:config)[:rancher_on_eks])
+    rescue Exception
+      # don't crash if a config isn't present (necessary for packaging)
+      puts("WARNING: config.yml not present - the application will not be usable without it.")
```

Comment:
if the application is not usable, should this not be an error ?
and should the application stops after informing ?

Reply:
We use the Rails stack to create an empty database for the package, and to bundle the assets. Neither is possible if the config is required.

Reply:
I added a config as requirement in the specfile: 23e2a29 .